pkg/sql/schemachanger/testdata/alter_table_add_column

```diff
@@ -400,81 +408,89 @@ upsert descriptor #106
   -    - 3
   -    - 4
   -    - 5
+  -    - 6
   -    targets:
   -    - elementProto:
-  -        column:
-  -          columnId: 2
-  -          defaultExpr: 42:::INT8
-  -          familyName: primary
-  -          pgAttributeNum: 2
+  -        primaryIndex:
+  -          embeddedIndex:
+  -            indexId: 1
+  -            keyColumnDirections:
+  -            - ASC
+  -            keyColumnIds:
+  -            - 1
+  -            tableId: 106
+  -            unique: true
+  -      metadata:
+  -        sourceElementId: 1
+  -        subWorkId: 1
+  -      targetStatus: ABSENT
+  -    - elementProto:
+  -        indexName:
+  -          indexId: 1
+  -          name: tbl_pkey
   -          tableId: 106
-  -          type:
-  -            family: IntFamily
-  -            oid: 20
-  -            width: 64
   -      metadata:
   -        sourceElementId: 1
   -        subWorkId: 1
-  -      targetStatus: PUBLIC
+  -      targetStatus: ABSENT
   -    - elementProto:
-  -        columnName:
+  -        columnDefaultExpression:
   -          columnId: 2
-  -          name: j
+  -          embeddedExpr:
+  -            expr: 42:::INT8
   -          tableId: 106
   -      metadata:
   -        sourceElementId: 1
   -        subWorkId: 1
   -      targetStatus: PUBLIC
   -    - elementProto:
-  -        primaryIndex:
-  -          indexId: 2
-  -          keyColumnDirections:
-  -          - ASC
-  -          keyColumnIds:
-  -          - 1
-  -          shardedDescriptor: {}
-  -          sourceIndexId: 1
-  -          storingColumnIds:
-  -          - 2
+  -        column:
+  -          columnId: 2
+  -          embeddedTypeT:
+  -            type:
+  -              family: IntFamily
+  -              oid: 20
+  -              width: 64
   -          tableId: 106
-  -          unique: true
   -      metadata:
   -        sourceElementId: 1
   -        subWorkId: 1
   -      targetStatus: PUBLIC
   -    - elementProto:
-  -        indexName:
-  -          indexId: 2
-  -          name: tbl_pkey
+  -        columnName:
+  -          columnId: 2
+  -          name: j
   -          tableId: 106
   -      metadata:
   -        sourceElementId: 1
   -        subWorkId: 1
   -      targetStatus: PUBLIC
   -    - elementProto:
   -        primaryIndex:
-  -          indexId: 1
-  -          keyColumnDirections:
-  -          - ASC
-  -          keyColumnIds:
-  -          - 1
-  -          shardedDescriptor: {}
-  -          sourceIndexId: 1
-  -          tableId: 106
-  -          unique: true
+  -          embeddedIndex:
```

Comment:
Can be follow-up but it feels like we ought to override the yaml marshalling for these to elide the `elementProto:` and `embeddedIndex`. It's noise.

Reply:
Agreed as a follow-up.